metrics/test/features/dashboard/presentation/state/project_metrics_store_test.dart

```diff
@@ -1,112 +1,137 @@
+import 'dart:async';
+
 import 'package:metrics/features/dashboard/domain/entities/build_metrics.dart';
 import 'package:metrics/features/dashboard/domain/entities/build_number_metric.dart';
 import 'package:metrics/features/dashboard/domain/entities/build_result_metric.dart';
-import 'package:metrics/features/dashboard/domain/entities/coverage.dart';
 import 'package:metrics/features/dashboard/domain/entities/performance_metric.dart';
-import 'package:metrics/features/dashboard/domain/usecases/get_build_metrics.dart';
-import 'package:metrics/features/dashboard/domain/usecases/get_project_coverage.dart';
-import 'package:metrics/features/dashboard/domain/usecases/parameters/project_id_param.dart';
+import 'package:metrics/features/dashboard/domain/entities/project.dart';
+import 'package:metrics/features/dashboard/domain/usecases/parameters/build_metrics_loading_param.dart';
+import 'package:metrics/features/dashboard/domain/usecases/receive_build_metrics_updates.dart';
+import 'package:metrics/features/dashboard/domain/usecases/receive_poject_updates.dart';
+import 'package:metrics/features/dashboard/presentation/model/project_metrics.dart';
 import 'package:metrics/features/dashboard/presentation/state/project_metrics_store.dart';
+import 'package:rxdart/rxdart.dart';
 import 'package:test/test.dart';
 
 void main() {
   const projectId = 'projectId';
-  const projectIdParam = ProjectIdParam(projectId: projectId);
-  const GetCoverageTestbed getCoverage = GetCoverageTestbed();
-  const GetBuildMetricsTestbed getBuildMetrics = GetBuildMetricsTestbed();
+  const ReceiveBuildMetricsUpdates receiveBuildMetricsUpdates =
+      ReceiveBuildMetricsUpdatesTestbed();
+  final ReceiveProjectUpdates receiveProjectUpdates =
+      ReceiveProjectUpdatesTestbed();
+
+  const BuildMetricsLoadingParam buildMetricsParams =
+      BuildMetricsLoadingParam(projectId, Duration(days: 5), 3);
+
   BuildMetrics expectedBuildMetrics;
 
   ProjectMetricsStore projectMetricsStore;
 
   setUpAll(() async {
     projectMetricsStore = ProjectMetricsStore(
-      getCoverage,
-      getBuildMetrics,
+      receiveProjectUpdates,
+      receiveBuildMetricsUpdates,
     );
 
-    expectedBuildMetrics = await getBuildMetrics(projectIdParam);
-    await projectMetricsStore.getCoverage(projectId);
-    await projectMetricsStore.getBuildMetrics(projectId);
+    expectedBuildMetrics =
+        await receiveBuildMetricsUpdates(buildMetricsParams).first;
+
+    await projectMetricsStore.subscribeToProjects();
+  });
+
+  test("Throws an assert if one of the use cases is null", () {
+    final assertionMatcher = throwsA(const TypeMatcher<AssertionError>());
```

Comment:
How about replacing this with the `throwsAssertionError` built-in matcher?

Reply:
I think it would be better to don't import the flutter_test package into unit tests to make them available even without a flutter.

Reply:
Sorry, I didn't notice that - nice catch! Then I'd like to suggest to use `isA<AssertionError>()` matcher instead of TypeMatcher directly. Thanks!